server/reports/commission/commission-report.js

```diff
@@ -26,50 +28,79 @@ export function handleCommissionRptRequest(request, response) {
+      rawReportData = reportTemp;
+      //console.log("CommissionReport.generate() returned this data: " + JSON.stringify(rawReportData));
+      return MindbodyAccess.getAuth();
+    }).then((authToken) => {
+      MindbodyAccess.authToken = authToken.AccessToken;
+      return MindbodyAccess.getStaff({StaffIds: Object.keys(rawReportData)});
+    }).then((getStaffResponse) => {
+      //console.log("getStaff returned: " + JSON.stringify(getStaffResponse));
+
+      //lookup staff names from the IDs returned by CommissionReport
+      let staffIdToNameDict = {};
+      staffIdToNameDict[CommissionReport.keyForSalesWithNoPriorInstructor] = "- Sales with no prior instructor -";
+      for (let staffInfo in getStaffResponse.StaffMembers) {
+        staffIdToNameDict[staffInfo.Id] = staffInfo.FirstName + " " + staffInfo.LastName;
+      }
 
+      //generate output
+      if (format === "csv") {
+        let csvRows = [];
+        let totalCommission = 0.0;
+        for (let staffId in rawReportData) {
+          //the value for rawReportData[staffId] is a dict of commission amounts that we must now total
+          let commission = 0.0;
+          for (let item in rawReportData[staffId]) {
+            commission += rawReportData[staffId][item];
+          }
+          csvRows.push({
+            "Instructor": staffIdToNameDict[staffId],   // staffId,
+            "Commission": commission,
+          });
+          //if (!(staffId === CommissionReport.keyForSalesWithNoPriorInstructor)) {
+            //todo: exclude SalesWithNoPriorInstructor? The requirements aren't clear here
+          //}
+          totalCommission += commission
+        }
+        // Additionally, add an extra row for totals
+        csvRows.push({
+            "Instructor": "Total",
+            "Commission": totalCommission,
+        });
 
-  // Generate output
-  if (format === "csv") {
-    let fields = ["paramName", "value"];
-    let parser = new json2csvExports.Parser({fields});
-    let csv = parser.parse(rptJsonRowsAsJson);
-    let fileName = "CommissionReport.csv";
-    //TODO: As a user, I'd appreciate a more unique filename with the date range embedded
-    response.setHeader("Content-Disposition", "attachment ; filename=\"" + fileName + "\"");
-    response.contentType("text/csv");
-    response.send(csv);
-  } else if (format === "json") {
-    response.json(rptJsonRowsAsJson);
+        let fields = ["Instructor", "Commission"];
+        let parser = new json2csvExports.Parser({fields});
+        let csv = parser.parse(csvRows);
+        let fileName = "CommissionReport.csv";
+        //todo: As a user, I'd appreciate a more unique filename with the date range embedded
+        response.setHeader("Content-Disposition", "attachment ; filename=\"" + fileName + "\"");
+        response.contentType("text/csv");
+        response.send(csv);
+      } else if (format === "json") {
+        response.json(rawReportData);
+      }
+    }).catch(error => {
```

Comment:
Like the comments say here, the main error we need to catch is the "1000 requests" one. Let me know if you know a better way to do that.